### DISCUSSION TITLE
How to use `validate()` with `AdvanceOrder[]`?

### DISCUSSION THREAD

**Discussion by valerioleo** [ProjectOpenSea/seaport#611] [Q&A]
As per title, I am curious on how to (possibly) validate `AdvancedOrder`s  instead of just `Order`s. 
I have a smart contract that is creating orders and storing them via `validate()` function, but they need to include some `extraData` field that will be used by the Zone to validate the execution. 

Using the `zoneHash` field would not be possible as data needs to be received in clear by the Zone.

**Comment by ryanio** [comment]
Advanced orders are validated against the zone with `isValidOrderIncludingExtraData`

**Reply by valerioleo** [reply]
As far as I understand, also Basic Orders can be validated by a Zone (via `isValidOrder`). Why can Basic Orders be validated via `validate()` function and Advanced Orders cannot?
